src/components/chips/js/chipsController.js

```diff
@@ -128,9 +183,13 @@
    */
   MdChipsCtrl.prototype.getChipBuffer = function() {
     if (this.mdAutocompleteCtrl) {
-      this.$log.error('md-autocomplete not yet supported');
-    } else if (this.hasInputElement) {
-      this.$log.error('user-provided inputs not yet supported');
+      // Should not be called from here.
+    } else if (this.userInputElement) {
+      if (this.userInputNgModelCtrl) {
+        return this.userInputNgModelCtrl.$viewValue;
+      } else {
+        return this.userInputElement[0].value;
+      }
     } else {
       return this.chipBuffer;
     }
```

Comment:
This version is much more clear:


```js
if (this.mdAutocompleteCtrl) {
  throw Error('...');
}

return !this.userInputElement ? this.chipsBuffer :
		this.userInputNgModelCtrl ? this.userInputNgModelCtrl.$viewValue : 
      	this.userInputElement[0].value;
```

Almost always best to use single-return at bottom of function.

> sometimes may use return at top for **rapid exits**; aka `returning early`